flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/runtimefilter/util/RuntimeFilterUtils.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.runtime.operators.runtimefilter.util;
+
+import org.apache.flink.core.memory.MemorySegmentFactory;
+import org.apache.flink.runtime.operators.util.BloomFilter;
+import org.apache.flink.table.data.GenericRowData;
+import org.apache.flink.table.data.RowData;
+
+import javax.annotation.Nullable;
+
+/** Utilities for runtime filter. */
+public class RuntimeFilterUtils {
+
+    public static final int OVER_MAX_ROW_COUNT = -1;
+
+    public static BloomFilter createOnHeapBloomFilter(int numExpectedEntries, double fpp) {
+        int byteSize = (int) Math.ceil(BloomFilter.optimalNumOfBits(numExpectedEntries, fpp) / 8D);
+        final BloomFilter filter = new BloomFilter(numExpectedEntries, byteSize);
+        filter.setBitsLocation(MemorySegmentFactory.allocateUnpooledSegment(byteSize), 0);
+        return filter;
+    }
+
+    public static RowData convertBloomFilterToRowData(
+            int actualRowCount, @Nullable BloomFilter bloomFilter) {
```

Comment:
-> The actRowCount use long type may be more suitable?
It makes sense. I will add a separate commit to do that.

-> Otherwise, it's not easy to understand what the value of actualRowCount is when BloomFilter is null.
The actualRowCount is `OverMaxRowCount ` iff `BloomFilter == null`. We have to send the OverMaxRowCount and null filter to downstream so that we can let the global runtime filter builder know that is already over-max-row-count.